extension/js/common/message-renderer.ts

```diff
@@ -431,7 +431,13 @@ export class MessageRenderer {
 
   private getMessageInfo = async (fullMsg: GmailRes.GmailMsg): Promise<MessageInfo> => {
     const sentDate = GmailParser.findHeader(fullMsg, 'date');
-    const sentDateStr = sentDate ? Str.fromDate(new Date(sentDate)).replace(' ', ' at ') : '';
+    const gmailDateReceived = $('div.gK span[title]').attr('title');
+    let sentDateStr: string;
+    if (gmailDateReceived !== undefined && !isNaN(new Date(gmailDateReceived).getTime())) {
+      sentDateStr = gmailDateReceived.replace(', ', ' at ');
+    } else {
+      sentDateStr = sentDate ? Str.fromDate(new Date(sentDate)).replace(' ', ' at ') : '';
+    }
```

Comment:
Why did you decide not to use `toLocaleString`? It seemed to work well - produced correctly formatted time strings.

Current implementation can lead to incorrect string for some users, as we don't exactly know which format they have in Gmail UI. For example how it formats date for me:

gmail ui:
<img width="200" alt="Screenshot 2024-04-26 at 12 10 02" src="https://github.com/FlowCrypt/flowcrypt-browser/assets/1062093/bebc1d95-e9fa-4fb1-98d0-fdc7548130aa">

print view (`Mar 14 at 2024` looks strange):
<img width="165" alt="Screenshot 2024-04-26 at 12 10 24" src="https://github.com/FlowCrypt/flowcrypt-browser/assets/1062093/259161eb-1cf1-4347-8f17-06a0c5dd1969">

Reply:
Hello @sosnovsky, thanks for checking it. Here are my comments.

> Why did you decide not to use toLocaleString? It seemed to work well - produced correctly formatted time strings.

I haven't gathered more information about why the original devs come up to using "toLocaleString()" so I thought that although I have used it correctly, there might be reason why they used it -- I have that in mind which give me hesitation -- but I'm looking into adding it now as I check it -- I definitely return correct formatted time.

> Current implementation can lead to incorrect string for some users, as we don't exactly know which format they have in Gmail UI. For example how it formats date for me:

This is a bit of a problem. Some date formats that Gmail shows isn't correctly recognized by javascript's "Date" constructor that leads to inaccurate dates. Would it be good if we used the dates from Gmail as is (safest option) or should be parse them? Parsing with other language could also have some drawbacks.

Reply:
I'll find a way... I believe I can.

Reply:
> I haven't gathered more information about why the original devs come up to using "toLocaleString()" so I thought that although I have used it correctly, there might be reason why they used it 

Probably `toISOString` is used for consistent formatting across different locales, but we've missed to apply timezone offset for it to have correct time. Here is possible solution - https://stackoverflow.com/a/28149561/3091318, let's add it to `Str.fromDate`

> This is a bit of a problem. Some date formats that Gmail shows isn't correctly recognized by javascript's "Date" constructor that leads to inaccurate dates. Would it be good if we used the dates from Gmail as is (safest option) or should be parse them? Parsing with other language could also have some drawbacks.

I think the most stable solution will be to use date from `sentDate` (like it was before), but for print string use `toLocaleString` instead of `Str.fromDate`

Reply:
Hello @sosnovsky , In my previous commit, I used "toLocaleString" inside "Str.fromDate" and now I have both issues fixed at the same time. So please let me know if you would the current over to the suggestion above?